src/Appwrite/Platform/Tasks/Restore.php

```diff
@@ -0,0 +1,291 @@
+            '--decompress',
+            '--strict',
+            '--remove-original', // Removes *.lz4 compressed files
+            '--parallel=' . self::PROCESSORS,
+            '--compress-threads=' . self::PROCESSORS,
+            '--target-dir=' . $target,
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec appwrite-xtrabackup xtrabackup ' . implode(' ', $args);
+        Backup::log($cmd);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+        Backup::log($stderr);
+
+        if (!str_contains($stderr, 'completed OK!') || !file_exists($target . '/xtrabackup_checkpoints')) {
+            Console::error('Decompress failed');
+            Console::exit();
+        }
+    }
+
+    public function prepare(string $target)
+    {
+        if (!file_exists($target)) {
+            Console::error('prepare error directory not found: ' . $target);
+            Console::exit();
+        }
+
+        $logfile = $target . '/../prepare.log';
+
+        $args = [
+            '--user=' . $this->dsn->getUser(),
+            '--password=' . $this->dsn->getPassword(),
+            '--host=' . $this->dsn->getHost(),
+            '--prepare',
+            '--strict',
+            '--target-dir=' . $target,
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec appwrite-xtrabackup xtrabackup ' . implode(' ', $args);
+        Backup::log($cmd);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+        Backup::log($stderr);
+
+        if (!str_contains($stderr, 'completed OK!') || !file_exists($target . '/xtrabackup_checkpoints')) {
+            Console::error('Prepare failed');
+            Console::exit();
+        }
+    }
+
+    public function restore(string $target, bool $cloud, string $datadir)
+    {
+        if (!file_exists($target)) {
+            Console::error('restore error directory not found: ' . $target);
+            Console::exit();
+        }
+
+        $logfile = $target . '/../restore.log';
+
+        $args = [
+            '--user=' . $this->dsn->getUser(),
+            '--password=' . $this->dsn->getPassword(),
+            '--host=' . $this->dsn->getHost(),
+            $cloud ? '--move-back' : '--copy-back',
+            '--strict',
+            '--target-dir=' . $target,
+            '--datadir=' . $datadir,
+            '--parallel=' . self::PROCESSORS,
+            '2> ' . $logfile,
+        ];
+
+        $cmd = 'docker exec appwrite-xtrabackup xtrabackup ' . implode(' ', $args);
+        Backup::log($cmd);
+        shell_exec($cmd);
+
+        $stderr = shell_exec('tail -1 ' . $logfile);
+        Backup::log($stderr);
+
+        if (!str_contains($stderr, 'completed OK!') || !file_exists($target . '/xtrabackup_checkpoints')) {
+            Console::error('Restore failed');
+            Console::exit();
+        }
+
+        // todo: Do we need to chown -R mysql:mysql /var/lib/mysql?
```

Comment:
Can you test if we need this? Otherwise let's remove the comment

Reply:
At the end of the move-back step, I see some files owner are root and some MySQL.
the MySQL container started with no error.
Trying to change own I get an error:
chown: unknown user/group mysql:mysql
not sure what to do

Reply:
As discussed we will leave this as is for now